File: problems/747-largest-number-at-least-twice-of-others/solution.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max = 0;
        int maxIndex = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > max) {
                max = nums[i];
                maxIndex = i;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (i != maxIndex && max < 2 * nums[i]) {
                return -1;
            }
        }
        return maxIndex;
    }
}

File: problems/747-largest-number-at-least-twice-of-others/solution.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        max_val = max(nums)
        max_index = nums.index(max_val)
        for i, num in enumerate(nums):
            if i != max_index and max_val < 2 * num:
                return -1
        return max_index